src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -649,8 +649,8 @@ private BoundExpression BindSimpleBinaryOperator(BinaryExpressionSyntax node, Di
                 {
                     resultSignature = signature;
                     HashSet<DiagnosticInfo> useSiteDiagnostics = null;
-                    bool leftDefault = left.IsLiteralDefault();
-                    bool rightDefault = right.IsLiteralDefault();
+                    bool leftDefault = left.IsLiteralDefaultOrTypelessNew();
```

Comment:
>left.IsLiteralDefaultOrTypelessNew(); [](start = 39, length = 37)

:memo: let's make sure that the speclet captures the rules around `new()` and comparisons (when is object equality valid?)